pygame_gui/core/colour_parser.py

```diff
@@ -0,0 +1,682 @@
+"""
+colour_parser.py
+A Functional Based Module for the parsing of Colour Strings in pygame_gui
+
+Use Notes:
+    Mostly all that one will ever need is these 5 functions 
+        - **parse_colour_or_gradient_string**: Attempt to parse a string that may be a colour or gradient into its respective value
+        - **is_valid_colour_string**: Check if a string represents a valid pygame Color
+        - **is_valid_gradient_string**: Check if a string represents a valid ColourGradient
+        - **parse_colour_string**: Parse a string into a pygame Color
+        - **parse_gradient_string**: Parse a string into a ColourGradient
+        
+    The documentation for what counts as a 'Valid' colour and gradient string can be found in the Theme Guide of the pygame_gui documentation at https://pygame-gui.readthedocs.io/en/v_065/theme_guide.html
+
+Developer Notes:
+    How it Works:
+        This module works through pairs of validating and parsing functions at the value, colour, and gradient levels
+        Hopefully everything should be generic enough that any new colour models or representations can be easily added without much bloat or *magic*
+
+        Parsing A Color:
+            Simply, The parsing of a color is done by simply checking if there is any valid function cashed in the _colourParsers dictionary, then calling the paired parsing function to get its value
+            Therefore, this system should be extremely extensible to add any 2 functions that can validate and parse a developer-determined schema
+
+        Parsing A Gradient:
+            As of now, the gradient parser is implemented in such a way where 
```

Comment:
Looks like you might have missed off the end of this sentence here.

Reply:
Thanks, added some implementation details for how the parser figures out which commas represent separations in the gradient and which commas represent separations in a colour's value. It could be important if someone wanted to add a colour schema that worked weirdly with the gradient separation